examples/run_storm_wiki_deepseek.py

```diff
@@ -0,0 +1,152 @@
+"""
+STORM Wiki pipeline powered by DeepSeek models and You.com or Bing search engine.
+You need to set up the following environment variables to run this script:
+    - DEEPSEEK_API_KEY: DeepSeek API key
+    - DEEPSEEK_API_BASE: DeepSeek API base URL (default is https://api.deepseek.com)
+    - YDC_API_KEY: You.com API key; or, BING_SEARCH_API_KEY: Bing Search API key
+
+Output will be structured as below
+args.output_dir/
+    topic_name/  # topic_name will follow convention of underscore-connected topic name w/o space and slash
+        conversation_log.json           # Log of information-seeking conversation
+        raw_search_results.json         # Raw search results from search engine
+        direct_gen_outline.txt          # Outline directly generated with LLM's parametric knowledge
+        storm_gen_outline.txt           # Outline refined with collected information
+        url_to_info.json                # Sources that are used in the final article
+        storm_gen_article.txt           # Final article generated
+        storm_gen_article_polished.txt  # Polished final article (if args.do_polish_article is True)
+"""
+
+import os
+import sys
+import re
+from argparse import ArgumentParser
+
+sys.path.append('./src')
+from lm import DeepSeekModel
+from rm import YouRM, BingSearch
+from storm_wiki.engine import STORMWikiRunnerArguments, STORMWikiRunner, STORMWikiLMConfigs
+from utils import load_api_key
+
+
+def sanitize_topic(topic):
+    """
+    Sanitize the topic name for use in file names.
+    Remove or replace characters that are not allowed in file names.
+    """
+    # Replace spaces with underscores
+    topic = topic.replace(' ', '_')
+
+    # Remove any character that isn't alphanumeric, underscore, or hyphen
+    topic = re.sub(r'[^a-zA-Z0-9_-]', '', topic)
+
+    # Ensure the topic isn't empty after sanitization
+    if not topic:
+        topic = "unnamed_topic"
+
+    return topic
+
+
+def main(args):
+    load_api_key(toml_file_path='secrets.toml')
+    lm_configs = STORMWikiLMConfigs()
+
+    # Ensure DEEPSEEK_API_KEY is set
+    if not os.getenv("DEEPSEEK_API_KEY"):
+        raise ValueError("DEEPSEEK_API_KEY environment variable is not set. Please set it in your secrets.toml file.")
+
+    deepseek_kwargs = {
+        'api_key': os.getenv("DEEPSEEK_API_KEY"),
+        'api_base': os.getenv("DEEPSEEK_API_BASE", "https://api.deepseek.com"),
+        'temperature': args.temperature,
+        'top_p': args.top_p,
+    }
+
+    # DeepSeek offers two main models: 'deepseek-chat' for general tasks and 'deepseek-coder' for coding tasks
```

Comment:
Hi @rmcc3 , thanks for the contribution! It looks pretty good.

Could you provide an input-output example (i.e., the generated article) using `deepseek-chat` or `deepseek-coder`? In this way, we will be more confident to merge this PR. Thank you so much!

Reply:
Sure thing!

Google was ran with Deepseek chat, and Stackoverflow was ran with Deepseek coder.
[Google.zip](https://github.com/user-attachments/files/16287525/Google.zip)
[Stackoverflow.zip](https://github.com/user-attachments/files/16287528/Stackoverflow.zip)

<img width="1367" alt="google" src="https://github.com/user-attachments/assets/5a714b2d-56aa-46fc-9300-52a5440c317a">
<img width="1367" alt="Stackoverflow" src="https://github.com/user-attachments/assets/da83a4d6-a77c-422b-a346-2c6d390119a2">